src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -108,7 +108,12 @@ public sealed class CosmosFhirDataStore : IFhirDataStore, IProvideCapability
             _modelInfoProvider = modelInfoProvider;
         }
 
-        public async Task<UpsertOutcome> UpsertAsync(
+        public async Task<UpsertOutcome> UpsertAsync(ResourceWrapperExtended resource, CancellationToken cancellationToken)
```

Comment:
Low priority: add checks to avoid null references of ResourceWrapperExtended.

Reply:
As you can see, I kept the same behavior as on original Upsert. Are you saying existing code is bad?